project_weather_app/main.py

```diff
@@ -0,0 +1,41 @@
+from database import WeatherDatabase
+from data_handler import DataHandler
+from data_service import DataServiceFactory
+from location import Location
+
+MODE = "API" # data retrieval mode: ["API"|"MOCK"]
+LONGITUDE = -122.431297 # "longitude": -122.431297,
+LATITUDE = 37.773972    # "latitude": 37.773972,
+
+DATA_SOURCE = "config.json" # config file for data service
+API_DB_FILE = "weather_api.db" # database file location
+MOCKED_DB_FILE = "weather_mocked.db" 
+
+def main():
+    print("Begin program.")
+
+    # select database file based on the mode
+    if MODE.upper() == "API":
+        print("Database: ", API_DB_FILE)
+        db_file = API_DB_FILE
+    elif MODE.upper() == "MOCK":
+        print("Database: ", MOCKED_DB_FILE)
+        db_file = MOCKED_DB_FILE
+    else:
+        raise("invalid mode")
```

Comment:
Suggestion to expand here, don't need to do right now, but you can define your own `Exception`s in something like an `exception.py`, which raise specific exceptions that are relevant to your project. I use that a lot in my projects to avoid raising generic exceptions.

Reference: https://martinxpn.medium.com/custom-exceptions-in-python-creating-custom-exceptions-59-100-days-of-python-4f26de8e851d

Reply:
Agreed, I use it heavily in my day-to-day life. Specifically, with different error codes which can be communicated to a front-end which they can translate to a specific error message, very useful.